services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -106,17 +107,55 @@ async def list_datasets(self, user_id: UserID) -> list[DatasetMetaData]:
             ]
 
     async def list_files_in_dataset(
-        self, user_id: UserID, dataset_id: str
+        self, user_id: UserID, dataset_id: str, expand_dirs: bool
     ) -> list[FileMetaData]:
         data: list[FileMetaData] = await self.list_files(
-            user_id, uuid_filter=dataset_id + "/"
+            user_id, expand_dirs, uuid_filter=ensure_ends_with(dataset_id, "/")
```

Comment:
ok I was wondering where this *ensure_ends_with* method was used. For me it makes much more sense to put this in the s3_client.py for the following reasons:
- the trick of the ```/``` at the end of the prefix is an S3-related bug/feature (the client should not know anything about this)
- in the s3_client it is centralized